### PR TITLE
Upload files by url

### PR DESCRIPTION
**Pull request by TrySound**
Here splitted asset upload into 2 phases

1. initial upload creates asset in db with uploading state and generates unique name (POST /rest/assets endpoint)

2. use unique name to upload file, extract meta and mark asset as uploaded (POST /rest/assets/$name endpoint)

This give us more straightforward data flow without creating temporary asssets with fake id which are changed later.

And also unlock splitting files into separate table to allow securely copy paste assets between projects (still wip).

Fixed persistent fetcher to support sequential submits.

## Code Review

- [ ] hi @istarkov, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/routes/rest.assets.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  try {
+    if (request.method === "POST") {
+      const formData = await request.formData();
+      const projectId = formData.get("projectId") as string;
+      const assetId = formData.get("assetId") as string;
+      const filename = formData.get("filename") as string;
+      if (projectId === null || assetId === null || filename === null) {
+        throw Error("Project id, asset id or filename are missing");
+      }
+      const name = await initiateUpload(
```
Comment (istarkov) [inline apps/builder/app/routes/rest.assets.tsx:36]: I dislike the name as its not upload but name getter, 
something similar to `presigned url`, as only this name allows to upload file later.
`createPresignedFileName` or something better

File: packages/asset-uploader/src/upload.ts [hunk 2]
```diff
@@ -1,30 +1,125 @@
...
+  { name, request }: { name: string; request: Request },
   client: AssetClient
 ) => {
-  const asset = await createAssetWithLimit(
-    projectId,
-    () => {
-      return client.uploadFile(request);
+  const asset = await prisma.asset.findFirst({
+    where: {
+      name: { equals: name },
```
Comment (istarkov) [inline packages/asset-uploader/src/upload.ts:91]: IMO you must add @unique constraint to the Asset table on `name`

Comment (TrySound) [reply packages/asset-uploader/src/upload.ts:91]: Not really. Name can be reused for cloned projects. Though I will make name an id in File table.

File: apps/builder/app/routes/rest.assets.$name.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  if (params.name === undefined) {
+    throw new Error("Name is undefined");
+  }
+
+  try {
+    if (request.method === "POST") {
+      const asset = await completeUpload(
+        {
+          name: params.name,
```
Comment (istarkov) [inline apps/builder/app/routes/rest.assets.$name.tsx:21]: What if everything in URL is considered as not secure, so bad actor can intercept and upload other photo with knowledge of `name`
Not an issue at all anyway as I have no idea why anyone would do that, does POST urls can leak as intents on mobile. And s3 presigned urls have no other security too

**Pull request merged**